lib/timer.js

```diff
@@ -418,13 +425,17 @@ timers.success('2e2f6dad-9678-4caf-bc41-8e62ca07d551', error)
   }
 
   disconnect(cb = noop) {
-    this[storage].close(noop)
+    this[storage].close(noop);
     this.transports[shutdown](() => {
+      const client = this.nats;
       this.nats.publish('skyring:node', JSON.stringify({
         node: this[kNode]
       , type: EVENT_STATUS.SHUTDOWN
-      }), noop)
-      setImmediate(this.nats.quit, cb)
+      }), function() {
+        client.flush(function() {
+          setImmediate(client.quit, cb);
+        });
+      });
```

Comment:
Flushing out the nats before closing fixes messages from being lost.

Reply:
Yep, this makes sense 👍